src/backends/torch/torchdataset.cc

```diff
@@ -311,4 +372,131 @@ namespace dd
     new_dataset._batches.insert(new_dataset._batches.end(), start_it, stop_it);
     return new_dataset;
   }
+
+  /*-- image tools --*/
+  int TorchDataset::add_image_file(const std::string &fname, const int &target,
+                                   const int &height, const int &width)
+  {
+    ImgTorchInputFileConn *inputc
+        = reinterpret_cast<ImgTorchInputFileConn *>(_inputc);
+
+    DDImg dimg;
+    inputc->copy_parameters_to(dimg);
+
+    try
+      {
+        if (dimg.read_file(fname))
+          {
+            this->_logger->error("Uri failed: {}", fname);
+          }
+      }
+    catch (std::exception &e)
+      {
+        this->_logger->error("Uri failed: {}", fname);
+      }
+    if (dimg._imgs.size() != 0)
+      {
+        at::Tensor imgt = image_to_tensor(dimg._imgs[0], height, width);
+        at::Tensor targett = target_to_tensor(target);
+
+        add_batch({ imgt }, { targett });
+        return 0;
+      }
+    else
+      {
+        return -1;
+      }
+  }
+
+  int TorchDataset::add_image_file(const std::string &fname,
```

Comment:
linked to comment below: here you pass _height and _width from upper inputc , but adding image needs other params from inputc that will be read through _dataset._inputc